c/src/ml-api-service-remote.c

```diff
@@ -378,6 +383,7 @@ _mlrs_process_remote_service (nns_edge_data_h data_h)
         ret = NNS_EDGE_ERROR_UNKNOWN;
       }
       g_byte_array_free (array, TRUE);
+      event_type = ML_SERVICE_EVENT_MODEL_REGISTERED;
```

Comment:
For each line (event_type = ML_SERVICE_EVENT_xx) we should check the result.
```
if (_mlrs_model_register (service_key, data_h, array->data, array->len)) { //done
  event_type = ML_SERVICE_EVENT_MODEL_REGISTERED;
} else { // failed, unnecessary to invoke the callback.
  _ml_error_report ("Failed to register model downloaded from: %s.",
      (gchar *) data);
  ret = NNS_EDGE_ERROR_UNKNOWN;
}
```

Reply:
Thanks, updated!